refinery/core/models.py

```diff
@@ -2213,3 +2220,87 @@ class SiteProfile(models.Model):
+    total_visualization_launches = models.IntegerField()
+    total_workflow_launches = models.IntegerField()
+    users_created = models.IntegerField()
+    unique_user_logins = models.IntegerField()
+
+    class Meta:
+        verbose_name_plural = "site statistics"
+
+    def save(self, force_insert=False, force_update=False, using=None,
+             update_fields=None):
+        self.run_date = timezone.now()
+        self.datasets_uploaded = self._get_datasets_uploaded().count()
+        self.datasets_shared = len(self._get_datasets_shared())
+        self.groups_created = self._get_number_of_groups_created()
+        self.users_created = self._get_users_created().count()
+        self.unique_user_logins = self._get_unique_user_logins().count()
+        self.total_user_logins = self._get_number_of_total_user_logins()
+        self.total_workflow_launches = \
+            self._get_total_workflow_launches().count()
+        self.total_visualization_launches = \
+            self._get_total_visualization_launches().count()
+        super(SiteStatistics, self).save()
+
+    def _get_previous_instance(self):
+        previous_instance = SiteStatistics.objects.filter(
+            run_date__lt=self.run_date
+        ).order_by('-run_date').first()
+
+        # Return the previous instance by run_date, or the current instance
+        # if a prior one doesn't exist
+        return previous_instance or self
+
+    def _get_datasets_shared(self):
+        return [
+            dataset for dataset in DataSet.objects.filter(
+                modification_date__gte=self._get_previous_instance().run_date
+            ) if dataset.shared
+        ]
+
+    def _get_datasets_uploaded(self):
+        return DataSet.objects.filter(
+            creation_date__gte=self._get_previous_instance().run_date
+        )
+
+    def _get_total_visualization_launches(self):
+        return tool_manager.models.VisualizationTool.objects.filter(
+            creation_date__gte=self._get_previous_instance().run_date
+        )
+
+    def _get_total_workflow_launches(self):
+        return tool_manager.models.WorkflowTool.objects.filter(
+            creation_date__gte=self._get_previous_instance().run_date
+        )
+
+    def _get_unique_user_logins(self):
+        return User.objects.filter(
+            last_login__gte=self._get_previous_instance().run_date
+        )
+
+    def _get_users_created(self):
+        return User.objects.filter(
+            date_joined__gte=self._get_previous_instance().run_date
+        )
+
+    def _get_number_of_groups_created(self):
+        return ExtendedGroup.objects.exclude(manager_group=None).count() - sum(
+            [s.groups_created for s in SiteStatistics.objects.all()]
+        )
+
+    def _get_number_of_total_user_logins(self):
+        return sum([u.login_count for u in UserProfile.objects.all()]) - \
+               sum([s.total_user_logins for s in SiteStatistics.objects.all()])
```

Comment:
I think you can just use a generator here and above: you don't need to make a comprehension from it:
```
sum(i*2 for i in range(3))
6
```